include/albatross/src/models/sparse_gp.hpp

```diff
@@ -646,8 +646,13 @@ auto rebase_inducing_points(
       fit_model.get_model().get_covariance()(new_inducing_points);
   new_fit.train_covariance = Eigen::SerializableLDLT(K_zz);
 
-  const JointDistribution new_prediction =
+  JointDistribution new_prediction =
       fit_model.predict(new_inducing_points).joint();
+  // We're going to need to take the sqrt of the new covariance which
+  // could be extremely small, so here we add a small nugget to avoid
+  // numerical instability
+  new_prediction.covariance.diagonal() += Eigen::VectorXd::Constant(
```

Comment:
will this have impact on the positive-definite property of the posteriori variance?

Reply:
It's actually helping ensure that the result is positive definite in situations where it may have ended up numerically negative.